### PR TITLE
[FEAT] Add template of webserv config format

### LINKED ISSUE #22: [TODO] Design config format
provide all feature config file

### PR DESCRIPTION
**Pull request by LeaYeh**
Resolved #22 

In the nginx_reference.conf, I noted how nginx config those requirements.
And in webserv_all_features.conf, I based on the NGINX to design the prototype for our project.

Please go through the each item to check it matches the subject.

Depends on the subject, in the configuration file, you should be able to
- [x] Choose the port and host of each ’server’
- [x] Setup the server_names or not
- [x] The first server for a host:port will be the default for this host:port
    - that means it will answer to all the requests that don’t belong to an other server
- [x] Setup default error pages
- [x] Limit client body size
- Setup routes with one or multiple of the following rules/configuration (routes wont be using regexp)
    - [x] Define a list of accepted HTTP methods for the route
    - [x] Define a HTTP redirection
    - [x] Define a directory or a file from where the file should be searched
        - for example, if url /kapouet is rooted to /tmp/www, url /kapouet/pouic/toto/pouet is /tmp/www/pouic/toto/pouet
    - [x] Turn on or off directory listing
    - [x] Set a default file to answer if the request is a directory
    - [x] Execute CGI based on certain file extension
    - [x] Make it work with POST and GET methods
    - [x] Make the route able to accept uploaded files and configure where they should be saved

### CODE REVIEW

**Review by yenthing — APPROVED** [review]
"The first server for a host:port will be the default for this host:port", because:
- _Default Server:_ The first server block that matches a given **listen** directive is treated as the default server for that particular host:port. NGINX will serve requests with this server block when no other block matches the request's Host header.
- _server_name:_ If a request comes in and the Host header doesn't match any server_name values specified, the default server will handle the request.

**Comment by LeaYeh** [comment]
"The first server for a host:port will be the default for this host:port", for example:

In this configuration, the first server block (associated with example.com) is considered the default server because it’s the first one defined.
```config
server {
    listen 80;
    server_name example.com;
    root /var/www/example.com;
}

server {
    listen 80;
    server_name other.com;
    root /var/www/other.com;
}
```

suppose an HTTP request comes in like this:

```http
GET / HTTP/1.1
Host: unknown.com
```

In this request, the Host header is set to unknown.com. However, unknown.com is not explicitly defined in your server configuration. Since the server doesn’t have a specific match for unknown.com, it will be handled by the default server, which is the first one you defined (example.com).

So, the request will be processed by the default server block (example.com) and will serve content from the /var/www/example.com directory, even though the request was made for unknown.com.

In summary, if a request doesn’t match any explicitly defined server blocks, it will be handled by the default server.

**Comment by yenthing** [comment]
Got cha
